sdk/go/common/util/gitutil/git_test.go

```diff
@@ -536,4 +536,34 @@ func TestParseAuthURL(t *testing.T) {
 		assert.NotNil(t, auth)
 		assert.Equal(t, "http-basic-auth - foo:<empty>", auth.String())
 	})
+
+	t.Run("Don't cache on error", func(t *testing.T) {
+		t.Parallel()
+		// Regresses https://github.com/pulumi/pulumi/issues/16637
+		originalPassphrase := os.Getenv(env.GitSSHPassphrase.Var().Name())
+		originalSocket := os.Getenv("SSH_AUTH_SOCK")
+		defer func() {
+			_ = os.Setenv(env.GitSSHPassphrase.Var().Name(), originalPassphrase)
+			_ = os.Setenv("SSH_AUTH_SOCK", originalSocket)
+		}()
+
+		err := os.Setenv(env.GitSSHPassphrase.Var().Name(), "incorrect passphrase")
+		require.NoError(t, err)
+
+		err = os.Unsetenv("SSH_AUTH_SOCK")
+		require.NoError(t, err)
```

Comment:
Can [`t.Setenv`](https://pkg.go.dev/testing#T.Setenv) be used to avoid having to manually set it back to the old values?

Should we not use `t.Parallel()` since the env vars affect the whole process?

Reply:
Yes, I blindly copied the test setup from another test without thinking too much. Then I blindly added the `nolint:paralleltest` after the linter complained. Clearly it was too late in the day when I was adding the tests 😭 

I updated all the tests in the file to use `t.Setenv` while I was at it https://github.com/pulumi/pulumi/pull/16652/commits/9869d58d0249d78faa67a32bc9747c86a999df0e